src/routes/teams.rs

```diff
@@ -1,25 +1,275 @@
+            .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+        if team_member.is_some() {
+            let team_members: Vec<crate::models::teams::TeamMember> = members_data
+                .into_iter()
+                .map(|data| crate::models::teams::TeamMember {
+                    user_id: data.user_id.into(),
+                    name: data.name,
+                    role: data.role,
+                    permissions: data.permissions,
+                })
+                .collect();
+
+            return Ok(HttpResponse::Ok().json(team_members));
+        }
+    }
+
     let team_members: Vec<crate::models::teams::TeamMember> = members_data
         .into_iter()
         .map(|data| crate::models::teams::TeamMember {
             user_id: data.user_id.into(),
             name: data.name,
             role: data.role,
+            permissions: Permissions::default(),
         })
         .collect();
 
     Ok(HttpResponse::Ok().json(team_members))
 }
+
+#[post("{id}/join")]
+pub async fn join_team(
+    req: HttpRequest,
+    info: web::Path<(TeamId,)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+
+    let mut transaction = pool
+        .begin()
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    // Edit Team Member to set Accepted to True
+    TeamMember::edit_team_member(
+        team_id,
+        current_user.id.into(),
+        None,
+        None,
+        Some(true),
+        None,
+        &mut transaction,
+    )
+    .await?;
+
+    transaction
+        .commit()
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    Ok(HttpResponse::Ok().body(""))
+}
+
+fn default_role() -> String {
+    "Member".to_string()
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct NewTeamMember {
+    pub user_id: UserId,
+    #[serde(default = "default_role")]
+    pub role: String,
+    #[serde(default = "Permissions::default")]
+    pub permissions: Permissions,
+}
+
+#[post("{id}/members")]
+pub async fn add_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId,)>,
+    pool: web::Data<PgPool>,
+    new_member: web::Json<NewTeamMember>,
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+
+    let mut transaction = pool
+        .begin()
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+    let team_member =
+        TeamMember::get_from_user_id(team_id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        if member.permissions.contains(Permissions::MANAGE_INVITES)
+            && member.accepted
```

Comment:
This condition should be removed.  `get_from_user_id` can only return accepted team members.